File: app/src/main/java/tw/tonyyang/englishwords/MainActivity.java
package tw.tonyyang.englishwords;

import android.app.Fragment;
import android.app.FragmentManager;
import com.google.android.material.tabs.TabLayout;
import androidx.legacy.app.FragmentStatePagerAdapter;
import androidx.viewpager.widget.ViewPager;
import androidx.appcompat.widget.Toolbar;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Arrays;
import java.util.List;

public class MainActivity extends BaseActivity {
    private static final Logger logger = LoggerFactory.getLogger(MainActivity.class);

    @Override
    protected void initToolbar() {
        super.initToolbar();
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
    }

    @Override
    protected int getLayoutResource() {
        return R.layout.activity_main;
    }

    @Override
    protected void onViewCreated() {
        final TabLayout tabs = findViewById(R.id.tabs);
        final ViewPager viewPager = findViewById(R.id.viewpager);
        tabs.removeAllTabs();
        tabs.addTab(tabs.newTab());
        tabs.addTab(tabs.newTab());
        tabs.addTab(tabs.newTab());
        viewPager.setAdapter(new SamplePagerAdapter(getFragmentManager(), tabs.getTabCount()));
        tabs.setupWithViewPager(viewPager);
    }

    private static class SamplePagerAdapter extends FragmentStatePagerAdapter {
        private int numOfTabs;
        private List<String> tabsTitle = Arrays.asList("讀取 / 更新", "單字列表", "單字考試");

        SamplePagerAdapter(FragmentManager fm, int numOfTabs) {
            super(fm);
            this.numOfTabs = numOfTabs;
        }

        @Override
        public CharSequence getPageTitle(int position) {
            return position < tabsTitle.size() ? tabsTitle.get(position) : null;
        }

        @Override
        public Fragment getItem(int position) {
            switch (position) {
                case 0:
                    return new FileChooserFragment();
                case 1:
                    return new WordListM1Fragment();
                case 2:
                    return new ExamFragment();
            }
            return null;
        }

        @Override
        public int getCount() {
            return numOfTabs;
        }
    }
}

File: app/src/main/java/tw/tonyyang/englishwords/MainActivity.kt
package tw.tonyyang.englishwords

import android.app.Fragment
import android.app.FragmentManager
import androidx.appcompat.widget.Toolbar
import androidx.legacy.app.FragmentStatePagerAdapter
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : BaseActivity() {
    override fun initToolbar() {
        super.initToolbar()
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
    }

    override val layoutResource: Int
        get() = R.layout.activity_main

    override fun onViewCreated() {
        tabs.removeAllTabs()
        tabs.addTab(tabs.newTab())
        tabs.addTab(tabs.newTab())
        tabs.addTab(tabs.newTab())
        viewpager.adapter = SamplePagerAdapter(fragmentManager, tabs.tabCount)
        tabs.setupWithViewPager(viewpager)
    }

    private class SamplePagerAdapter internal constructor(fm: FragmentManager?, private val numOfTabs: Int) : FragmentStatePagerAdapter(fm) {
        private val tabsTitle = Arrays.asList("讀取 / 更新", "單字列表", "單字考試")
        override fun getPageTitle(position: Int): CharSequence? {
            return if (position < tabsTitle.size) tabsTitle[position] else null
        }

        override fun getItem(position: Int): Fragment? {
            when (position) {
                0 -> return FileChooserFragment()
                1 -> return WordListM1Fragment()
                2 -> return ExamFragment()
            }
            return null
        }

        override fun getCount(): Int {
            return numOfTabs
        }
    }
}